### PR TITLE
nQueens solution for distributed computing

### PR DESCRIPTION
**Pull request by andyttran**

### CODE REVIEW

File: server/nqueens.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
+//An distributed computing version of nqueens solver
+function findNQueensParallel(n) {
+  
+  //Estimate total time in milliseconds based on trial runs
+  var est = 4.9*Math.pow(10, -6)*Math.pow(2.71, 1.6*n);
```
Comment (ekong2) [inline server/nqueens.js]: what program did you use to curve fit this?

File: server/nqueens.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  var dvv = {};
+
+  //Create chucks to distribute to slaves
+  dvv.partition = createChunks(n, rows);
+
+  //Minified version of nQueens function below
+  dvv.func = function Q(r,i){var n=0;if(void 0===r)return n;var a=function(r,f,t,v){v===i&&n++;for(var o=0;i>o;o++)if(!r[o]){var u=o-v,c=o+v;f[u]||t[c]||(f[u]=!0,t[c]=!0,r[o]=!0,v++,a(r,f,t,v),f[u]=!1,t[c]=!1,r[o]=!1,v--)}};return a.apply(this,r),n};
+
+  //Callback function sums up all solutions found
+  dvv.callback = function(results){
```
Comment (ekong2) [inline server/nqueens.js]: this might need to be modified

**Comment by ekong2** [comment]
does partitioning takes as much time as solving it? both have similar logic structures

File: server/nqueens.js (hunk 1 above)
Comment (andyttran) [inline server/nqueens.js]: I used an exponential fit in MATLAB

**Comment by andyttran** [comment]
The partitioning stops when keys get to  'rows' length, this is much faster than finding a whole solution

File: server/nqueens.js (hunk 2 above)
Comment (andyttran) [inline server/nqueens.js]: agreed, the callback will have to fit well with the server side, and func may need to be stringified.

**Comment by ekong2** [comment]
did you test how much time it takes?

**Comment by andyttran** [comment]
The partition time takes just 3ms for a N(17) case with 2 row partitioning.

**Pull request merged**